src/containers/BooksForm.js

```diff
@@ -10,25 +12,55 @@ const categories = [
   'Sci-Fi',
 ];
 
-const BooksForm = () => (
-  <form>
-    <h4> Title:</h4>
-    <input
-      type="text"
-      name="title"
-      value="{title}"
-    />
-    <h4>Category:</h4>
-    <select
-      name="category"
-      value=""
-    >
-      {categories.map((category) => (
-        <option key={category}>{category}</option>
-      ))}
-    </select>
-    <button type="submit">Submit</button>
-  </form>
-);
+const BooksForm = () => {
+  const [book, setBook] = useState({ title: '', category: '' });
+
+  let title = '';
+  let category = '';
+  const dispatch = useDispatch();
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+
+    if (name === 'title') {
+      title = value;
+      setBook({ ...book, title });
+    }
+    if (name === 'category') {
+      category = value;
+      setBook({ ...book, category });
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(addBook({
+      ...book,
+      id: Math.random().toString(36).substring(7),
+    }));
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <h4> Title:</h4>
+      <input
+        name="title"
+        type="text"
+        onChange={(e) => handleChange(e)}
+      />
+      <h4>Category:</h4>
+      <select
+        name="category"
+        value="{category}"
+        onChange={(e) => handleChange(e)}
+      >
+        {categories.map((category) => (
+          <option key={category}>{category}</option>
```

Comment:
- [ ] [OPTIONAL] I cannot choose any category when I want to create book. The value of the select option doesn't change. Please fix it.

Reply:
It works on my end.

Reply:
For some reason, it only doesn't work for the first book created.

Reply:
It only ignores the first category selected. I don't know how to fix that.